p5/sketch/Skia2DRenderer/image.py

```diff
@@ -0,0 +1,118 @@
+from p5.core import constants
+from p5.core import PImage
+import numpy as np
+import skia
+import builtins
+
+
+class SkiaPImage(PImage):
+    def __init__(self, width, height, pixels=None):
+        self._width = width
+        self._height = height
+        self.pixels = (
+            pixels
+            if pixels is not None
+            else np.zeros((width, height, 4), dtype=np.uint8)
+        )
+
+    @property
+    def width(self):
+        return self._width
+
+    @width.setter
+    def width(self, width):
+        self._width = width
+
+    @property
+    def height(self):
+        return self._height
+
+    @height.setter
+    def height(self, height):
+        self._height = height
+
+    @property
+    def size(self):
+        return (self.width, self.height)
+
+    @size.setter
+    def size(self, size):
+        self.width, self.height = size
+        self.pixels.resize((*size, 4))
+
+    @property
+    def aspect_ratio(self):
+        return self.width / self.height
+
+    def load_pixels(self):
+        builtins.pixels = self.pixels
+
+    def update_pixels(self):
+        pass
+
+    def mask(self, image):
+        """
+        :param image: Image to be used as make
+        :type image: SkiaPImage
+        """
+        with skia.Surface(self.pixels) as canvas:
+            canvas.drawImage(image.surface.makeImageSnapshot())
+
+    def filter(self, kind, param=0.5):
+        if kind == constants.THRESHOLD:
+            threshold = param * 255
+            mask = np.dot(self.pixels[..., :3], [0.2989, 0.5870, 0.1140]) < threshold
```

Comment:
Could you add a reference to these numbers?

Reply:
https://www.baeldung.com/cs/convert-rgb-to-grayscale#3-luminosity-method

This method is being used everywhere to convert RGB images to GREY

Reply:
Ah I see. Thank you!